core/trino-main/src/test/java/io/trino/sql/planner/optimizations/TestAddLocalExchangesForTaskScaleWriters.java

```diff
@@ -0,0 +1,125 @@
+        queryRunner.createCatalog("mock_report_written_bytes", createConnectorFactorySupportingReportingBytesWritten(true, "mock_report_written_bytes"), ImmutableMap.of());
+        return queryRunner;
+    }
+
+    private MockConnectorFactory createConnectorFactorySupportingReportingBytesWritten(boolean supportsWrittenBytes, String name)
+    {
+        return MockConnectorFactory.builder()
+                .withSupportsReportingWrittenBytes(supportsWrittenBytes)
+                .withGetTableHandle(((session, schemaTableName) -> null))
+                .withName(name)
+                .build();
+    }
+
+    @Test
+    public void testLocalScaledWriterDistributionWithSupportsReportingWrittenBytes()
+    {
+        // when task scale writers is enabled
+        assertDistributedPlan(
+                "CREATE TABLE mock_report_written_bytes.mock.test AS SELECT * FROM tpch.tiny.nation",
+                testSessionBuilder()
+                        .setSystemProperty(TASK_SCALE_WRITERS_ENABLED, "true")
+                        .setSystemProperty(SCALE_WRITERS, "false")
+                        .build(),
+                anyTree(
+                        exchange(LOCAL, REPARTITION, SCALED_WRITER_DISTRIBUTION,
+                                anyTree(
+                                        tableScan("nation")))));
+
+        // when task scale writers is disabled
+        assertDistributedPlanDoesNotMatch(
+                "CREATE TABLE mock_report_written_bytes.mock.test AS SELECT * FROM tpch.tiny.nation",
+                testSessionBuilder()
+                        .setSystemProperty(TASK_SCALE_WRITERS_ENABLED, "false")
+                        .setSystemProperty(SCALE_WRITERS, "false")
+                        .build(),
+                anyTree(
+                        exchange(LOCAL, REPARTITION, SCALED_WRITER_DISTRIBUTION,
+                                anyTree(
+                                        tableScan("nation")))));
+    }
+
+    @Test
+    public void testLocalScaledWriterDistributionWithoutSupportsReportingWrittenBytes()
+    {
+        // when task scale writers is enabled
+        assertDistributedPlanDoesNotMatch(
+                "CREATE TABLE mock_dont_report_written_bytes.mock.test AS SELECT * FROM tpch.tiny.nation",
+                testSessionBuilder()
+                        .setSystemProperty(TASK_SCALE_WRITERS_ENABLED, "true")
+                        .setSystemProperty(SCALE_WRITERS, "false")
+                        .build(),
+                anyTree(
+                        exchange(LOCAL, REPARTITION, SCALED_WRITER_DISTRIBUTION,
+                                anyTree(
+                                        tableScan("nation")))));
+
+        // when task scale writers is disabled
+        assertDistributedPlanDoesNotMatch(
```

Comment:
ah, this is confusing. Why can't we match exact plan?

Reply:
It's super easy to mismatch on something else than intended in these tests. Hence exact match checks are more resilient